src/NHibernate/Cfg/SettingsFactory.cs

```diff
@@ -393,14 +393,16 @@ private static System.Type CreateLinqQueryProviderType(IDictionary<string, strin
 		private static ITransactionFactory CreateTransactionFactory(IDictionary<string, string> properties)
 		{
 			string className = PropertiesHelper.GetString(
-				Environment.TransactionStrategy, properties, typeof(AdoNetWithDistributedTransactionFactory).FullName);
+				Environment.TransactionStrategy, properties, typeof(AdoNetWithSystemTransactionFactory).FullName);
 			log.Info("Transaction factory: " + className);
 
 			try
 			{
-				return
+				var transactionFactory =
 					(ITransactionFactory)
 					Environment.BytecodeProvider.ObjectsFactory.CreateInstance(ReflectHelper.ClassForName(className));
+				transactionFactory.Configure(properties);
```

Comment:
The `Configure` was already there, but un-called and with an obsolete parameter type. Now it is used.